concent_api/core/subtask_helpers.py

```diff
@@ -185,3 +187,18 @@ def are_subtask_results_accepted_messages_signed_by_the_same_requestor(subtask_r
         ) for subtask_results_accepted in subtask_results_accepted_list
     )
     return are_all_signed_by_requestor
+
+
+def get_and_lock_subtask(
+    subtask_id: str,
+    conditions: Dict[str, Any] = None,
+) -> Optional[Subtask]:
+    assert 'subtask_id' not in conditions
```

Comment:
Maybe it would be better to
1. Remove `subtask_id` from parameters and this `assert`.
2. Assert that `subtask_id` key is in `conditions` and its value is a string.
@cameel What do you think ?

Or even better - I saw there was a discussion to make it more generic. How about maybe then making following parameters:
- `query_set_or_model` - as mentioned in @cameel comment
- `unique_id_field_name` - `subtask_id` in this case.
- `unique_id_field_type` - `str` in this case.
- `conditions` - `str` in this case.

It is just a proposal and please treat it as one. I don't want to hold this pull request due to all possible variations of how this function can be implemented (as it depends on how it will be used mostly).

Reply:
> Maybe it would be better to
> 1. Remove `subtask_id` from parameters and this `assert`.
> 2. Assert that `subtask_id` key is in `conditions` and its value is a string.
> @cameel What do you think ?

It's a good idea.

> Or even better - I saw there was a discussion to make it more generic. How about maybe then making following parameters:
> - `query_set_or_model` - as mentioned in @cameel comment
> - `unique_id_field_name` - `subtask_id` in this case.
> - `unique_id_field_type` - `str` in this case.
> - `conditions` - `str` in this case.

I still think that what I described in https://github.com/golemfactory/concent/pull/778#discussion_r215346045 is the way to go. It's very simple, flexible and reusable.

Reply:
done